tex/pgfcircshapes.tex

```diff
@@ -92,6 +92,23 @@
 		}
 }
 
+%% Diamond terminal
+
+\pgfdeclareshape{diamond}{
+	\anchor{center}{
+		\pgfpointorigin
+	}
+
+	\behindforegroundpath{		
+		\pgfscope
+			\pgftransformrotate{45}
+			\pgfpathrectanglecorners{\pgfpoint{\pgf@x+\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}{\pgf@y-\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}}{\pgfpoint{\pgf@x-\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}{\pgf@y+\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}}
```

Comment:
You can use the dimensions \pgf@circ@res@other, \pgf@circ@res@temp and \pgf@circ@res@step as temporary variables. To assign it with a mathematical expression use
\pgfmathsetlength, like e.g. here: https://github.com/circuitikz/circuitikz/blob/master/tex/pgfcircbipoles.tex#L364